packages/components/src/components/text/text.jsx

```diff
@@ -4,40 +4,30 @@ import PropTypes from 'prop-types';
 import { isEmptyObject } from '@deriv/shared';
 
 const Text = ({ children, size, color, align, weight, line_height, as, className, styles, ...props }) => {
-    const setStyle = () => {
+    const [style, setStyle] = React.useState({});
+    React.useEffect(() => {
         const class_styles = {
             '--text-size': `var(--text-size-${size || 's'})`,
             '--text-color': `var(--text-${color || 'general'})`,
             '--text-lh': `var(--text-lh-${line_height || 'm'})`,
             '--text-weight': `var(--text-weight-${weight || 'normal'})`,
+            '--text-align': `var(--text-align-${align || 'start'})`,
         };
         if (!isEmptyObject(styles)) {
-            return { ...class_styles, ...styles };
+            const combined_style = { ...class_styles, ...styles };
+            setStyle(combined_style);
         }
-        return class_styles;
-    };
+        setStyle(class_styles);
+    }, [size, color, line_height, weight]);
```

Comment:
You could just combine both `class_styles` and `styles` on each re-render (it's cheap). This will remove the unnecessary complexity (maintaining state) and it allows the consumer of this component to provide dynamic `styles` too.